In [1]:
import pandas as pd

In [2]:
titanic = pd.read_csv("C://Users//wkdal//Desktop//archive//titanic.csv")

In [3]:
titanic.head()

pclass  survived                                             name     sex  \
0     1.0       1.0                    Allen, Miss. Elisabeth Walton  female   
1     1.0       1.0                   Allison, Master. Hudson Trevor    male   
2     1.0       0.0                     Allison, Miss. Helen Loraine  female   
3     1.0       0.0             Allison, Mr. Hudson Joshua Creighton    male   
4     1.0       0.0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

       age  sibsp  parch  ticket      fare    cabin embarked boat   body  \
0  29.0000    0.0    0.0   24160  211.3375       B5        S    2    NaN   
1   0.9167    1.0    2.0  113781  151.5500  C22 C26        S   11    NaN   
2   2.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN    NaN   
3  30.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN  135.0   
4  25.0000    1.0    2.0  113781  151.5500  C22 C26        S  NaN    NaN   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [7]:
# 열 추가

titanic['name_new'] = ['minji jang']
titanic['age_new'] = [30]
titanic['Driver_new'] = [True]

ValueError: Length of values (1) does not match length of index (1310)

ValueError: Length of values (1) does not match length of index (1310)

열 추가 할 때, titanic 데이터프레임에는 1310개의 행(세로)가 있어서 나머지를 채울 수 없다

In [8]:
# 나머지는 null값으로 채우고 1개의 행만 아래의 값으로 채움
titanic['name_new'] =  pd.Series(['minji jang'], index=[0])
titanic['age_new'] =  pd.Series(['1'], index=[0])
titanic['Driver_new'] =  pd.Series(['30'], index=[0])

In [9]:
titanic.head(3)

pclass  survived                            name     sex      age  sibsp  \
0     1.0       1.0   Allen, Miss. Elisabeth Walton  female  29.0000    0.0   
1     1.0       1.0  Allison, Master. Hudson Trevor    male   0.9167    1.0   
2     1.0       0.0    Allison, Miss. Helen Loraine  female   2.0000    1.0   

   parch  ticket      fare    cabin embarked boat  body  \
0    0.0   24160  211.3375       B5        S    2   NaN   
1    2.0  113781  151.5500  C22 C26        S   11   NaN   
2    2.0  113781  151.5500  C22 C26        S  NaN   NaN   

                         home.dest    name_new age_new Driver_new  
0                     St Louis, MO  minji jang       1         30  
1  Montreal, PQ / Chesterville, ON         NaN     NaN        NaN  
2  Montreal, PQ / Chesterville, ON         NaN     NaN        NaN

✅ DataFrame 구조 요약
개념	방향	예시
행(row)	세로	사람 1명, 2명, 3명...
열(column)	가로	이름, 나이, 성별 같은 속성들